### PR TITLE
fixes for dae test

### LINKED ISSUE #670: DAE test broken
### What happened?

DAE test fails with both MCT and NUOPC

### What are the steps to reproduce the bug?

`./create_test DAE_Vnuopc.f45_f45_mg37.FHS94.izumi_nag.cam-dae`

### What CAM tag were you using?

cam6_3_078

### What machine were you running CAM on?

CISL machine (e.g. cheyenne)

### What compiler were you using?

Intel

### Path to a case directory, if applicable

_No response_

### Will you be addressing this bug yourself?

No

### Extra info

This test has been broken for a long time.
It is fixable in NUOPC.

### PR DESCRIPTION
**Pull request by mvertens**
This fixes #670 - but also needs an accompanying CMEPS PR https://github.com/ESCOMP/CMEPS/pull/314.
To have the DAE test pass, the CMEPS must point to cmeps0.13.74 or later. However, this PR can be merged without a  CMEPS PR as long as the DAE test is not expected to pass.

### CODE REVIEW

File: src/cpl/nuopc/atm_comp_nuopc.F90 [hunk 1]
```diff
@@ -558,6 +549,19 @@ subroutine InitializeRealize(gcomp, importState, exportState, clock, rc)
        call shr_sys_abort( subname//' ERROR: unknown start_type' )
     end if
 
+    ! DART always starts up as an initial run.
+    call NUOPC_CompAttributeGet(gcomp, name='data_assimilation_atm', value=cvalue, &
+         isPresent=isPresent, isSet=isSet, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    if (isPresent and isSet) then
```
Comment (nusbaume) [inline src/cpl/nuopc/atm_comp_nuopc.F90:556]: Is this valid syntax?  Or does it need to be:

```suggestion
    if (isPresent .and. isSet) then
```

**Review by nusbaume — COMMENTED** [review]
Looks good, but I did have one syntax-related question I wanted to ask before full approval.

File: src/cpl/nuopc/atm_comp_nuopc.F90 (hunk 1 above)
Comment (mvertens) [reply src/cpl/nuopc/atm_comp_nuopc.F90:556]: @nusbaume - thanks for catching this. Its a bug that I introduced. At the last minute I wanted to make this change backwards compatible without needing a corresponding change in CMEPS. So I added the ispresent and isset flags - and clearly did not try to compile. Sorry about that. 
Its very easy to add these attributes to cime_config in CMEPS and have the component check for it. That said - to actually have the DAE test pass - you do need the CMEPS tag that adds data_assimilation_atm to the config attributes that the atm receives.

Comment (nusbaume) [reply src/cpl/nuopc/atm_comp_nuopc.F90:556]: @mvertens Thanks for the explanation, and no worries on the bug!  I mostly just wanted to make sure I haven't been adding a bunch of unnecessary `.` to the logical operators all this time.

Are you ok with me pushing to your branch?  That way I can get this fixed and pass it along to the other SEs for review (after which we'll probably bring it in along with CMEPS update).

**Review by nusbaume — APPROVED** [review]

**Review by peverwhee — APPROVED** [review]
Thanks @mvertens ! I'll bring this into the upcoming (hopefully tomorrow) misc tag.

**Pull request merged**